File: Labs/Lab-04-20231013/app/src/main/java/sap/escooters/service_layer/ServiceLayer.java
package sap.escooters.service_layer;

import io.vertx.core.json.JsonObject;
import sap.layers.Layer;

public interface ServiceLayer extends Layer {
	
	/* about users */
	void registerNewUser(String id, String name, String surname) throws UserIdAlreadyExistingException;
	JsonObject getUserInfo(String id) throws UserNotFoundException;

	/* about escooters */
	void registerNewEScooter(String id) throws UserIdAlreadyExistingException;
	JsonObject getEScooterInfo(String id) throws EScooterNotFoundException;
	
	/* about rides */
	String startNewRide(String userId, String escooterId) throws RideNotPossibleException;
	JsonObject getRideInfo(String id) throws RideNotFoundException;
	void endRide(String rideId) throws RideNotFoundException, RideAlreadyEndedException;
}

File: Labs/Lab-05-20231016/app/src/main/kotlin/assigment03/escooters/service_layer/ServiceLayer.kt
package assigment03.escooters.service_layer

import assigment03.layers.Layer
import io.vertx.core.json.JsonObject

interface ServiceLayer : Layer {
    fun registerNewUser(id: String, name: String, surname: String)
    fun getUserInfo(id: String): JsonObject
    fun registerNewEScooter(id: String)
    fun getEScooterInfo(id: String): JsonObject
    fun startNewRide(userId: String, escooterId: String): String
    fun getRideInfo(id: String): JsonObject
    fun endRide(rideId: String)
}